packages/rest-typings/src/v1/omnichannel.ts

```diff
@@ -13,20 +13,778 @@ import type {
 	IRoom,
 	ISetting,
 } from '@rocket.chat/core-typings';
+import Ajv from 'ajv';
 
 import type { PaginatedRequest } from '../helpers/PaginatedRequest';
 import type { PaginatedResult } from '../helpers/PaginatedResult';
 
 type booleanString = 'true' | 'false';
 
+const ajv = new Ajv({
+	coerceTypes: true,
+});
+
+type LivechatVisitorsInfo = {
+	visitorId: string;
+};
+
+const LivechatVisitorsInfoSchema = {
+	type: 'object',
+	properties: {
+		visitorId: {
+			type: 'string',
+		},
+	},
+	required: ['visitorId'],
+	additionalProperties: false,
+};
+
+export const isLivechatVisitorsInfoProps = ajv.compile<LivechatVisitorsInfo>(LivechatVisitorsInfoSchema);
+
+type LivechatRoomOnHold = {
+	roomId: IRoom['_id'];
+};
+
+const LivechatRoomOnHoldSchema = {
+	type: 'object',
+	properties: {
+		roomId: {
+			type: 'string',
+		},
+	},
+	required: ['roomId'],
+	additionalProperties: false,
+};
+
+export const isLivechatRoomOnHoldProps = ajv.compile<LivechatRoomOnHold>(LivechatRoomOnHoldSchema);
+
+type LivechatDepartmentId = {
+	onlyMyDepartments?: booleanString;
+	includeAgents?: booleanString;
+};
+
+const LivechatDepartmentIdSchema = {
+	type: 'object',
+	properties: {
+		onlyMyDepartments: {
+			type: 'string',
+			nullable: true,
+		},
+		includeAgents: {
+			type: 'string',
+			nullable: true,
+		},
+	},
+	additionalProperties: false,
+};
+
+export const isLivechatDepartmentIdProps = ajv.compile<LivechatDepartmentId>(LivechatDepartmentIdSchema);
+
+type LivechatDepartmentAutocomplete = {
+	selector: string;
+	onlyMyDepartments: booleanString;
+};
+
+const LivechatDepartmentAutocompleteSchema = {
+	type: 'object',
+	properties: {
+		selector: {
+			type: 'string',
+		},
+		onlyMyDepartments: {
+			type: 'string',
+		},
+	},
+	required: ['selector', 'onlyMyDepartments'],
+	additionalProperties: false,
+};
+
+export const isLivechatDepartmentAutocompleteProps = ajv.compile<LivechatDepartmentAutocomplete>(LivechatDepartmentAutocompleteSchema);
+
+type LivechatDepartmentDepartmentIdAgentsGET = {
+	sort: string;
+};
+
+const LivechatDepartmentDepartmentIdAgentsGETSchema = {
+	type: 'object',
+	properties: {
+		sort: {
+			type: 'string',
+		},
+	},
+	required: ['sort'],
+	additionalProperties: false,
+};
+
+export const isLivechatDepartmentDepartmentIdAgentsGETProps = ajv.compile<LivechatDepartmentDepartmentIdAgentsGET>(
+	LivechatDepartmentDepartmentIdAgentsGETSchema,
+);
+
+type LivechatDepartmentDepartmentIdAgentsPOST = {
+	upsert: string[];
+	remove: string[];
+};
+
+const LivechatDepartmentDepartmentIdAgentsPOSTSchema = {
+	type: 'object',
+	properties: {
+		upsert: {
+			type: 'array',
+			items: {
+				type: 'string',
+			},
+		},
+		remove: {
+			type: 'array',
+			items: {
+				type: 'string',
+			},
+		},
+	},
+	required: ['upsert', 'remove'],
+	additionalProperties: false,
+};
+
+export const isLivechatDepartmentDepartmentIdAgentsPOSTProps = ajv.compile<LivechatDepartmentDepartmentIdAgentsPOST>(
```

Comment:
This kind of types would become more common, would be good to define an "standard" for it, like `GETLivechatDepartmentIdAgents` and `POSTLivechatDepartmentIdAgents`, prefixing the method to all typedefs